addon/helpers/tag-name-is-component.js

```diff
@@ -0,0 +1,7 @@
+import { helper } from '@ember/component/helper';
+
+function tagNameIsComponent([as]) {
+  return typeof as === 'object';
```

Comment:
do we want this to return true for things that aren't components?

Reply:
Technically no, but it would be weird for someone to pass an object that isn't a component.

Perhaps there are some other things we could check for on the object? If not, I think this is likely good enough for now.

Reply:
I mean, this is a quick win, we can make it more constrained later as the public API declares what is intended anyway